src/refactor_nrepl/ns/suggest_aliases.clj

```diff
@@ -28,6 +28,9 @@
         fragments (map take-last
                        (range 1 (inc (count fragments)))
                        (repeat (distinct fragments)))
+        fragments (remove (fn [segments]
+                            (re-find #"^\d+" (first segments)))
+                          fragments)
```

Comment:
```suggestion
```

Let's do the `remove` a few lines below:

```clj
(into {}
      (keep (fn [segments]
              (let [candidate (->> segments (string/join "."))]
                (when-not (invalid-symbol? candidate)
                  [(symbol candidate),
                   [namespace-name]]))))
      fragments)
```

where: 

```clj
(def invalid-symbol? (memoize (fn [x]
                                (try
                                  (edn/read-string x)
                                  false
                                  (catch Exception _
                                    true)))))
```

This way we gain genericity (my no. 1 concern - I don't want people to encounter similar problems in the future) and possibly performance.

Please make sure to unit-test `invalid-symbol?`

Thanks!

Reply:
(feel free to find a better name for `invalid-symbol?` - it accepts strings not symbols)

Reply:
Sure thing! I added a `symbol?` test too, since we probably shouldn't accept keywords etc either.